src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,296 @@
+    setDelayedLabel:React.Dispatch<React.SetStateAction<string | undefined>>;
+}
+
+interface ICommonEntryProps{
+    fontSize?:string;
+    bold?:boolean;
+    separatorWidth?:number;
+    editMode?:boolean;
+    margin?:string;
+    padding?:string;
+}
+
+interface IHeadingEntryProps extends ICommonEntryProps{
+    label:string;
+    justifyContent?:string;
+    secondLabel?:string;
+    infoRef?:React.MutableRefObject<IInfo | null>;
+    editable?:boolean;
+}
+
+const HeadingEntry=forwardRef<HTMLDivElement,IHeadingEntryProps>(({label,editMode,editable,separatorWidth=1,secondLabel,infoRef,...props},ref):React.ReactElement=>{
+    const [delayedLabel,setDelayedLabel]=useState<string>()
+    const [secondLabelState,setSecondLabelState]=useState(secondLabel)
+
+    /**
+     * this uploads the info to the parent component
+     */
+    useEffect(()=>{
+        if(infoRef) infoRef.current={setDelayedLabel}
+    },[])
+
+    /**
+     * updates state for second label based on event fired
+     * @param event {React.ChangeEvent<HTMLInputElement>} - the fired event 
+     */
+    const updateSecondLabel=useCallback(({target}:React.ChangeEvent<HTMLInputElement>)=>{
+        setSecondLabelState(target.value)
+    },[]) 
+
+    /**
+     * 
+     * @returns 
+     */
+    const renderSecondLabel=useCallback(()=>{
+        if(secondLabelState){
+            if(editMode&&editable)
+                return <Input value={secondLabelState} onChange={updateSecondLabel} width={HEADER_ENTRY_INPUT_WIDTH} />
+            return <div>{secondLabelState}</div>
+        }
+        return null
+    },[secondLabelState])
+
+    return (
+        <EntryContainer separatorWidth={separatorWidth} {...props} ref={ref}>
+            <div>{label||delayedLabel}</div>
+            {renderSecondLabel()}
+        </EntryContainer>
+    )
+})
+
+interface IEntryProps extends ICommonEntryProps{
+    amount?:number;
+    dailyAmount:number;
+    label:string;
+    unity:string;
+    indentationNumber?:number;
+    indentationSize?:number;
+}
+
+const Entry:React.FC<IEntryProps>=({
+    amount=0,
+    dailyAmount,
+    label,
+    unity,
+    editMode,
+    bold=false,
+    separatorWidth=1,
+    indentationNumber=0,
+    indentationSize=4,
+    ...props
+}):React.ReactElement=>{
+    const [amountState,setAmountState]=useState<string>(amount.toString())
+    
+    /**
+     * this computes the daily value percentage
+     */
+    const dailyValuePercentage= useMemo(()=>{
+        if(dailyAmount){
+            /**
+             * gets the percentage of a value respect a totalValue
+             * @param value {number} - value for which we want to get percentage
+             * @param totalValue {number} - value for which percentage is 100 
+             * @returns {number} the percentage
+             */
+            const getPercentage=(value:number,totalValue:number)=>Math.round(value/totalValue*100)
+            return getPercentage(parseInt(amountState,10),dailyAmount) 
```

Comment:
What and why?

Reply:
this is because in some place of edition I was using it in more than one place so I extracted logic of getting percentage into a function. later on I finally ended up only using in one place.

Reply:
Then fix it, don't leave it like this if your not reusing it